From a598cc994ee3ec7a89594b3504aaeecfba78a922 Mon Sep 17 00:00:00 2001
From: Jaka Bac <jakabac@gmail.com>
Date: Thu, 15 Oct 2026 16:01:12 +0200
Subject: [PATCH] avcodec/dnxhddata: fix incorrect luma and chroma weight table
 usage for CID 1271

(cherry picked from commit a598cc994ee3ec7a89594b3504aaeecfba78a922)
---
 libavcodec/dnxhddata.c     | 2 +-
 tests/ref/fate/dnxhr-12bit | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/libavcodec/dnxhddata.c b/libavcodec/dnxhddata.c
index ce06d77f18e..3d113abfdb9 100644
--- a/libavcodec/dnxhddata.c
+++ b/libavcodec/dnxhddata.c
@@ -1049,7 +1049,7 @@ static const CIDEntry dnxhd_cid_table[] = {
       { 0 }, { 57344, 255} },
     { 1271, DNXHD_VARIABLE, DNXHD_VARIABLE, DNXHD_VARIABLE, DNXHD_VARIABLE,
       0, 6, DNXHD_VARIABLE, 4,
-      dnxhd_1241_luma_weight, dnxhd_1241_chroma_weight,
+      dnxhd_1235_luma_weight, dnxhd_1235_chroma_weight,
       dnxhd_1235_dc_codes, dnxhd_1235_dc_bits,
       dnxhd_1235_ac_codes, dnxhd_1235_ac_bits, dnxhd_1235_ac_info,
       dnxhd_1235_run_codes, dnxhd_1235_run_bits, dnxhd_1235_run,
diff --git a/tests/ref/fate/dnxhr-12bit b/tests/ref/fate/dnxhr-12bit
index eb5716780cb..c6e8fd99ae2 100644
--- a/tests/ref/fate/dnxhr-12bit
+++ b/tests/ref/fate/dnxhr-12bit
@@ -3,4 +3,4 @@
 #codec_id 0: rawvideo
 #dimensions 0: 1920x1080
 #sar 0: 1/1
-0,          0,          0,        1,  8294400, 0x31bfa8f1
+0,          0,          0,        1,  8294400, 0xbe1dc546
-- 
2.53.0

